### PR TITLE
exceptions: Generate Error() method

### PR DESCRIPTION
**Pull request by abhinav**
Depends on #66.

### CODE REVIEW

File: gen/struct.go [hunk 1]
```diff
@@ -127,6 +127,17 @@ func structure(g Generator, spec *compile.StructSpec) error {
 				<$strings>.Join(<$fields>[:<$i>], ", "),
 			)
 		}
+
+		<if .Spec.IsExceptionType>
+			func (<$v> <$structRef>) Error() string {
+				return <$v>.String()
+			}
+
+			func Is<$structName>(err error) bool {
```
Comment (prashantv) [inline gen/struct.go]: don't need this  since it's just a cast. what we wanted was `Is<Method>Error`

Comment (abhinav) [inline gen/struct.go]: :+1:

**Comment by abhinav** [comment]
#66 is merged. Rebased on master.

**Comment by prashantv** [comment]
lgtm

**Pull request merged**